Source/JavaScriptCore/wasm/WasmAirIRGeneratorBase.h

```diff
@@ -3551,12 +3551,15 @@ Expected<std::unique_ptr<InternalFunction>, String> parseAndCompileAirImpl(Compi
     auto& procedure = *compilationContext.procedure;
     Code& code = procedure.code();
 
+    // FIXME: Workaround for clang on ARMv7: it tries to call the OpcodeOrigin despite the constexpr
+#if USE(JSVALUE64)
```

Comment:
This ifdef looks wrong. Can you investigate why?

Reply:
It seems like a gcc bug... The untaken branch in an if constexpr cannot be ill-formed. 

From https://en.cppreference.com/w/cpp/language/if:
```
Note: the discarded statement cannot be ill-formed for every possible specialization:

template<typename T>
void f()
{
    if constexpr ([std::is_arithmetic_v](http://en.cppreference.com/w/cpp/types/is_arithmetic)<T>)
        // ...
    else {
        using invalid_array = int[-1]; // ill-formed: invalid for every T
        static_assert(false, "Must be arithmetic"); // ill-formed before CWG2518
    }
}
```

Another solution would be to create the missing constructor with a `UNREACHABLE_FOR_PLATFORM();`, like it's done for `OpcodeOrigin::opcode()` and `OpcodeOrigin::location()`. WDYT?

Reply:
I got an answer from the clang developers: https://github.com/llvm/llvm-project/issues/60780

tl;dr: both clang and GCC are right here